module_analysis/1_calc_module_composition.py

```diff
@@ -0,0 +1,283 @@
+        # else:
+        #    out_df.loc[module_id, "module label"] = "mixed"
+
+    return out_df
+
+
+# %%time
+# Get labels for PAO1 compendium
+pao1_module_labels = label_modules(
+    pao1_module_ids, pao1_membership, pao1_core, pao1_acc, pao1_gene_group_composition
+)
+
+# %%time
+# Get labels of PA14 compendium
+pa14_module_labels = label_modules(
+    pa14_module_ids, pa14_membership, pa14_core, pa14_acc, pa14_gene_group_composition
+)
+
+pao1_module_labels.head()
+
+pa14_module_labels.head()
+
+# Distribution of p-values
+sns.displot(pao1_module_labels["p-value"])
+sns.displot(pa14_module_labels["p-value"])
+
+# ### Examine module composition
+
+pao1_module_labels["module label"].value_counts()
+
+pa14_module_labels["module label"].value_counts()
+
+# Add size of modules to df
+pao1_module_labels["size"] = pao1_membership["module id"].value_counts()
+pa14_module_labels["size"] = pa14_membership["module id"].value_counts()
+
+pao1_module_labels.head()
+
+pa14_module_labels.head()
+
+# +
+# Size distributions of PAO1 modules
+f1 = sns.displot(
+    pao1_module_labels.loc[pao1_module_labels["module label"] == "mixed", "size"]
+)
+plt.title("Size distribution of PAO1 mixed modules")
+
+f2 = sns.displot(
+    pao1_module_labels.loc[
+        pao1_module_labels["module label"] == "mostly accessory", "size"
+    ]
+)
+plt.title("Size distribution of PAO1 mostly accessory modules")
+
+f3 = sns.displot(
+    pao1_module_labels.loc[pao1_module_labels["module label"] == "mostly core", "size"]
+)
+plt.title("Size distribution of PAO1 mostly core modules")
+
+# +
+# Size distributions of PA14 moduels
+g1 = sns.displot(
+    pa14_module_labels.loc[pa14_module_labels["module label"] == "mixed", "size"]
+)
+plt.title("Size distribution of PA14 mixed modules")
+
+g2 = sns.displot(
+    pa14_module_labels.loc[
+        pa14_module_labels["module label"] == "mostly accessory", "size"
+    ]
+)
+plt.title("Size distribution of PA14 mostly accessory modules")
+
+g3 = sns.displot(
+    pa14_module_labels.loc[pa14_module_labels["module label"] == "mostly core", "size"]
+)
+plt.title("Size distribution of PA14 mostly core modules")
+# -
+
+# Save
+pao1_module_labels.to_csv(
+    os.path.join(paths.LOCAL_DATA_DIR, "pao1_gene_module_labels.tsv"), sep="\t"
+)
+pa14_module_labels.to_csv(
+    os.path.join(paths.LOCAL_DATA_DIR, "pa14_gene_module_labels.tsv"), sep="\t"
+)
+
+# +
+# TO DO: Compare the composition across partitions
+# Save the matrix that maps module id to module label per partition
+# In another notebook look through module labels per partition
+# Calculate consistency of labeling using ARI for each pair of partitions
+# -
+
+# **Takeaway:**
+# * Most modules are mixed, some are mostly accessory. Only PA14 compendium have some mostly core modules
+# * PAO1 mixed and mostly accessory modules have similar sizes (~10 genes)
+# * PA14 mixed modules are ~10 genes, mostly accessory modules are smaller ~5 genes, mostly core modules are larger ~ 20 genes
```

Comment:
You don't have any mostly core modules after multiple testing correction, right? Might be good to update this comment.